cpp/ql/test/experimental/query-tests/Security/CWE/CWE-552/semmle/tests/test.cpp

```diff
@@ -0,0 +1,174 @@
+#define NULL 0
+
+typedef struct {} FILE;
+typedef	unsigned int	uint_t; 
+typedef	uint_t	mode_t;	
+typedef unsigned long size_t;
+
+FILE *fdopen(int handle, char *mode);
+FILE *fopen(const char *pathname, const char *mode);
+int fclose(FILE *stream);
+int open(const char *pathname, int flags);
+int open(const char *pathname, int flags, mode_t mode);
+int close(int file);
+FILE *fdopen(int handle, char *mode);
+int rename(const char *from, const char *to);
+bool remove(const char *path);
+int fread(char *buf, int size, int count, FILE *fp);
+size_t fwrite(const void *ptr, size_t size, size_t nmemb, FILE *stream);
+
+void test1(const char *from,const char *to)
+{
+  FILE *f1 = NULL;
+  FILE *f2 = NULL;
+  char data[1000];
+  int count;
+  f1 = fopen(from, "r");
+  count = fread(data, 1, 1000, f1);
+  fclose(f1);
+  rename(from,to); // GOOD
```

Comment:
Why is this `GOOD`? The return value of `rename` is not checked.

Reply:
you are right the example is not successful, from the point of view of the correctness of work with `rename`.
I suggest deleting it.

Reply:
Please change this to `BAD` and ensure it's capture by your query if possible.